### PR TITLE
[Orchestrator] Collect conditions for k8s resources

### PR DESCRIPTION
**Pull request by JLineaweaver**
### What does this PR do?

Previously most resources used the most recent condition time and message. We might want to query on previous conditions so we want to start collecting all of them. This pr adds that for resources missing it

### Motivation

Greatly enhance query ability by creating more tags

### Additional Notes

Just need to make sure there are no missing nil checks

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/daemonset.go [hunk 1]
```diff
@@ -49,8 +52,41 @@ func ExtractDaemonSet(ds *appsv1.DaemonSet) *model.DaemonSet {
...
+			Status:  string(condition.Status),
+			Type:    string(condition.Type),
+		}
+		if !condition.LastTransitionTime.IsZero() {
+			c.LastTransitionTime = condition.LastTransitionTime.Unix()
+		}
+
+		conditions = append(conditions, c)
+
+		conditionTag := fmt.Sprintf("kube_condition_%s:%s", strings.ToLower(string(condition.Type)), strings.ToLower(string(condition.Status)))
```
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/daemonset.go:87]: nit: this line could be a common function so the tag format is only declared once

**Review by xlucas — COMMENTED** [review]
LGTM overall!
Could you also update tests for replicasets, daemonsets and statefulsets?

File: releasenotes/notes/collect-resource-conditions-84adbeaa73033b99.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - Collect conditions for a variety of kubernetes resources
```
Comment (rtrieu) [inline releasenotes/notes/collect-resource-conditions-84adbeaa73033b99.yaml:10]:
```suggestion
  - Collect conditions for a variety of Kubernetes resources.
```

**Review by rtrieu — APPROVED** [review]
Minor suggestion but approved.

**Review by kangyili — APPROVED** [review]
We need to add same release note for dca as well

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/daemonset.go (hunk 1 above)
Comment (JLineaweaver) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/daemonset.go]: [d7f518b](https://github.com/DataDog/datadog-agent/pull/16499/commits/d7f518bfc5d0fc86e364d6596fca4d9999c12cb3)

**Review by kangyili — APPROVED** [review]

**Pull request merged**